### PR TITLE
YONK-742: User update API allowed to set country,city & gender as NULL

### PR DESCRIPTION
**Pull request by aamishbaloch**
User update API allowed to set following attributes as NULL
- country
- city
- gender

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

File: edx_solutions_api_integration/users/views.py [hunk 1]
```diff
@@ -636,12 +636,17 @@ def post(self, request, user_id):  # pylint: disable=R0915
...
-            country = request.data.get('country')
-            if country:
-                existing_user_profile.country = country
+
+            existing_user_profile.city = request.data.get('city')
+            existing_user_profile.country = request.data.get('country')
+            existing_user_profile.gender = request.data.get('gender')
+
+            # Empty title is also allowed
+            existing_user_profile.title = request.data.get('title', existing_user_profile.title)
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: I think this appraoch should be applied to `city`, `country`, `gender` and unit tests should be updated to test scenarios where if `city`, `country`, `gender` and `title` is missing it should not be updated.

File: edx_solutions_api_integration/users/views.py [hunk 2]
```diff
@@ -652,17 +657,7 @@ def post(self, request, user_id):  # pylint: disable=R0915
                 # If they give us garbage, just ignore it instead
                 # of asking them to put an integer.
                 year_of_birth = None
-            if year_of_birth:
```
Comment (ziafazal) [inline edx_solutions_api_integration/users/views.py]: why `year_of_birth` is not being updated?

Comment (aamishbaloch) [inline edx_solutions_api_integration/users/views.py]: Its being updated in L660.

**Comment by aamishbaloch** [comment]
@ziafazal done with the suggested changes. Can you review the changes?

**Pull request merged**